src/pages/index.js

```diff
@@ -74,115 +88,266 @@ const closeModalOnEsc = (event) => {
+profileElements.editButton.addEventListener("click", () => {
+  profileElements.nameInput.value = profileElements.nameElement.textContent;
+  profileElements.jobInput.value = profileElements.jobElement.textContent;
+  openModal(profileElements.modal);
 });
 
-profileCloseButton.addEventListener("click", () => {
-  closeModal(editProfileModal);
+profileElements.closeButton.addEventListener("click", () => {
+  closeModal(profileElements.modal);
 });
 
-newPostButton.addEventListener("click", () => {
-  openModal(newPostModal);
+newPostElements.button.addEventListener("click", () => {
+  openModal(newPostElements.modal);
 });
 
-newPostCloseButton.addEventListener("click", () => {
-  closeModal(newPostModal);
+newPostElements.closeButton.addEventListener("click", () => {
+  closeModal(newPostElements.modal);
 });
 
-profileFormElement.addEventListener("submit", handleProfileFormSubmit);
+profileElements.form.addEventListener("submit", handleProfileFormSubmit);
+newPostElements.form.addEventListener("submit", handleNewPostFormSubmit);
+
+cardElements.modalCloseButton.addEventListener("click", () => {
+  closeModal(cardElements.modal);
+});
 
-newPostFormElement.addEventListener("submit", handleNewPostFormSubmit);
+cardElements.modalImage.addEventListener("click", () => {
+  closeModal(cardElements.modal);
+});
 
-// Form Submissions
+// Form Submission Handlers
 function handleProfileFormSubmit(event) {
   event.preventDefault();
 
-  profileNameElement.textContent = profileNameInput.value;
-  profileJobElement.textContent = profileJobInput.value;
+  const submitButton = profileElements.form.querySelector(
+    ".modal__submit-button"
+  );
+  const originalText = submitButton.textContent;
+  submitButton.textContent = "Saving...";
 
-  closeModal(editProfileModal);
+  api
+    .editUserInfo({
+      name: profileElements.nameInput.value,
+      about: profileElements.jobInput.value,
+    })
+    .then((data) => {
+      profileElements.nameElement.textContent = data.name;
+      profileElements.jobElement.textContent = data.about;
+      closeModal(profileElements.modal);
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      submitButton.textContent = originalText;
+    });
 }
 
 function handleNewPostFormSubmit(event) {
   event.preventDefault();
 
-  const newCardData = {
-    name: newPostprofileNameInput.value,
-    link: newPostLinkInput.value,
-  };
+  const submitButton = newPostElements.form.querySelector(
+    ".modal__submit-button"
+  );
+  const originalText = submitButton.textContent;
+  submitButton.textContent = "Saving...";
 
-  const cardElement = getCardElement(newCardData);
-  cardList.prepend(cardElement);
+  api
+    .addNewCard({
+      name: newPostElements.nameInput.value,
+      link: newPostElements.linkInput.value,
+    })
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardElements.list.prepend(cardElement);
```

Comment:
**Could be improved**


You can make a universal function for adding a card into the section using different methods: `append`, `prepend`, or anything else:

```js

// The function accepts a card object and a method of adding to the section
// The method is initially `prepend` but you can pass `append` 
function renderCard(item, method = "prepend") {
  
  const cardElement = getCardElement(item);
  // Add the card into the section using the method
  cardElements.list[ method ](cardElement);
}



```